Cesium.CodeGen/Ir/Expressions/FunctionCallExpression.cs

```diff
@@ -36,6 +36,35 @@ public FunctionCallExpression(Ast.FunctionCallExpression expression)
 
     public IExpression Lower(IDeclarationScope scope)
     {
+        if (_function.Identifier == "__builtin_offsetof")
+        {
+            if (_arguments.Count != 2 ||
```

Comment:
I think following cases not yet handled 
```
struct foo
{
    int bar;
    int baz;
    int qux;
}

offsetof(struct S, baz);
```

Maybe that require special tricks on the parser, create special construct for passing type name in the functions. Maybe easier target for learning would be implementing `sizeof()` in the parser first, because we already have https://github.com/ForNeVeR/Cesium/blob/6e6136c335f29a03afdd6354e4c0a7932e405029/Cesium.CodeGen/Ir/Expressions/SizeOfExpression.cs